mesonbuild/environment.py

```diff
@@ -1321,7 +1321,7 @@ def sanitize(p):
                 linker = self._guess_win_linker(['link'], cls, for_machine)
                 return cls(
                     compiler, version, for_machine, is_cross, info, exe_wrap,
-                    target, linker=linker)
+                    target, full_version=cl_signature, linker=linker)
```

Comment:
Hmm, what's this for? Is it for the same reason as https://github.com/mesonbuild/meson/pull/7021/commits/c01c4f01075f34ba22f6314473f5e44a4bfb8c53 for ARM64 detection?

Reply:
No.  I think full_version is only used in logging, so this is just so we can see that the cross-compiler is being used.